From:   Allen Pais <allen.lkml@gmail.com>
To:     linux-kernel@vger.kernel.org
Cc:     clm@fb.com, dsterba@suse.com, linux-btrfs@vger.kernel.org,
        Allen Pais <allen.lkml@gmail.com>
Subject: [PATCH v2] fs:btrfs: return -ENOMEM on allocation failure.
Date:   Wed, 20 Sep 2017 11:47:46 +0530

v2: return -ENOMEM from btrfsic_dev_state_alloc() too.

[HH: backported to 4.9.x]
Signed-off-by: Allen Pais <allen.lkml@gmail.com>
---
 fs/btrfs/check-integrity.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/fs/btrfs/check-integrity.c b/fs/btrfs/check-integrity.c
index 7d5a9b5..9db1e76 100644
--- a/fs/btrfs/check-integrity.c
+++ b/fs/btrfs/check-integrity.c
@@ -2913,8 +2913,8 @@ int btrfsic_mount(struct btrfs_root *root,
 	if (!state) {
 		state = vzalloc(sizeof(*state));
 		if (!state) {
-			pr_info("btrfs check-integrity: vzalloc() failed!\n");
-			return -1;
+			pr_info("btrfs check-integrity: allocation failed!\n");
+			return -ENOMEM;
 		}
 	}
 
@@ -2945,7 +2945,7 @@ int btrfsic_mount(struct btrfs_root *root,
 		if (NULL == ds) {
 			pr_info("btrfs check-integrity: kmalloc() failed!\n");
 			mutex_unlock(&btrfsic_mutex);
-			return -1;
+			return -ENOMEM;
 		}
 		ds->bdev = device->bdev;
 		ds->state = state;
